Reflection/test.cc

```diff
@@ -150,3 +159,297 @@ TEST(Reflection, CurrentStructInternals) {
   static_assert(std::is_same<SuperType<DerivedFromFoo>, Foo>::value, "");
   EXPECT_EQ(1u, FieldCounter<DerivedFromFoo>::value);
 }
+
+namespace reflection_test {
+
+CURRENT_STRUCT(Serializable) {
+  CURRENT_FIELD(i, uint64_t);
+  CURRENT_FIELD(s, std::string);
+};
+
+CURRENT_STRUCT(ComplexSerializable) {
+  CURRENT_FIELD(j, uint64_t);
+  CURRENT_FIELD(q, std::string);
+  CURRENT_FIELD(v, std::vector<std::string>);
+  CURRENT_FIELD(z, Serializable);
+};
+
+struct CollectFieldNames {
+  std::vector<std::string>& output_;
+  template <typename T>
+  void operator()(current::reflection::TypeSelector<T>, const std::string& name) const {
+    output_.push_back(name);
+  }
+};
+
+}  // namespace reflection_test
+
+TEST(Reflection, VisitAllFields) {
+  using namespace reflection_test;
+
+  EXPECT_EQ(
+      "struct Serializable {\n"
+      "  uint64_t i;\n"
+      "  std::string s;\n"
+      "};\n",
+      Reflector().DescribeCppStruct<Serializable>());
+  {
+    std::vector<std::string> result;
+    CollectFieldNames names{result};
+    current::reflection::VisitAllFields<Serializable, current::reflection::FieldTypeAndName>::WithoutObject(
```

Comment:
As for now, it could be just `current::reflection::FieldName`.
What do you think, is it worth to keep all the functions or just leave one - `FieldTypeAndName`?